### PR TITLE
Org Model: Add DomainGroup and Suborganization models

### LINKED ISSUE #2299: Add DomainGroup and Suborganization tables
### Issue description

For the new Org Model feature we need to have User Domain groups so that domains in an org can be given a label to effectively group them together. This label will be used for filtering domains for the org member. 

### Acceptance criteria

DomainGroup and Suborganization are added tables in django admin with the following

- [x] only viewable to superusers
- [x] have a `name` textfield ( for Domain group this will be the "label" tied to the domains for suborgs this is the sub-agency name such as CISA or FEMA )
- [x] has portfolio field that is a foreign key field to Portfolio
- [x] see [the domain group section](https://miro.com/app/board/uXjVKAEDJB4=/?moveToWidget=3458764591533149219&cot=14) and [sub organization](https://miro.com/app/board/uXjVKAEDJB4=/?moveToWidget=3458764591533149225&cot=14) of the architecture model. Double check that DomainGroup and sub org created matches the architecture there

Domain group also has the following:
- [x] `domains`, many to many field linked to DomainInfo

### Additional context

_No response_

### Links to other issues

blocked by #2248

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #2299 

## Changes

- Adds models for DomainGroup and Suborganization, both viewable to superusers on admin page.
- Additional quickfix updating Portfolio string representation to org name instead of portfolio object

## Setup

Go to[ getgov-es admin page](https://getgov-es.app.cloud.gov/admin/) and navigate to DomainGroup and Suborganization pages. You should be able to create, view, update, and delete any entries in the DomainGroup and Suborganization tables.
DomainGroup table
image

Suborganization table
image

DomainGroup and Suborganization models should not be accessible on analyst account
image

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [x] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/admin.py
Comment (abroddrick) [inline src/registrar/admin.py:2660]: @dave-kennedy-ecs should we add these for new models? (I am thinking not) This would require additional work to get them imported correct?

File: src/registrar/models/suborganization.py
Comment (abroddrick) [inline src/registrar/models/suborganization.py:11]: **change before merging** sub org shouldn't be null or blank. Also it's a character field we should have something as a max (max_length)  as a best practice. I would default to a high number, like 1000.

**Review by abroddrick — CHANGES_REQUESTED** [review]
Very quick change request, add a user group migration otherwise this won't show in django admin. Ping me when done and when that max_length is added, and I imagine it will take just a few mins for me to approve this

**Review by therealslimhsiehdy — APPROVED** [review]
LGTM both as as superuser (visible and everything interconnected) and confirmed that Analysts cannot view the changes. Would wait on merging though for a response for [this comment] (https://github.com/cisagov/manage.get.gov/pull/2345/files#r1648240875) though maybe for confirmation

File: src/registrar/models/domain_group.py
Comment (erinysong) [reply src/registrar/models/domain_group.py]: Totally makes sense and just updated it - thank you for the suggestion!

File: src/registrar/admin.py
Comment (abroddrick) [reply src/registrar/admin.py:2660]: what would determine if something needs to be included in the export/import tables? For instance, if we add a new value to domain requests/domain info that is an fk to another table, is that when we would need to update the import/export feature so that we don't break anything?

Comment (dave-kennedy-ecs) [reply src/registrar/admin.py:2660]: Yes, that is one condition that would require us to update import/export - anything new tables which have fk from a current table in the import/export set.
And, generally, any data that needs to be replicated in an export/import scenario would need to update the import/export scripts

**Comment by erinysong** [comment]
> Conditionally approving. Based on Dave's comments, it seems like we don't need the export on the Suborg and domain group tables yet, so I would just remove those until we are ready to export portfolio

Sounds good! Updated the PR to remove references to the Resource class in our added admin classes

**Pull request merged**
